velox/functions/sparksql/aggregates/tests/MinMaxAggregationTest.cpp

```diff
@@ -0,0 +1,299 @@
+    VectorFuzzer::Options options;
+    options.vectorSize = 1'000;
+    options.nullRatio = 0.1;
+    VectorFuzzer fuzzer(options, pool());
+    std::vector<RowVectorPtr> vectors(10);
+    for (auto i = 0; i < 10; ++i) {
+      vectors[i] = fuzzer.fuzzInputRow(rowType);
+    }
+    return vectors;
+  }
+
+  void doTest(const TypePtr& inputType, bool testWithTableScan = true) {
+    auto rowType = ROW({"c0", "c1", "mask"}, {BIGINT(), inputType, BOOLEAN()});
+    auto vectors = fuzzData(rowType);
+    createDuckDbTable(vectors);
+
+    static const std::string c0 = "c0";
+    static const std::string c1 = "c1";
+    static const std::string a0 = "a0";
+
+    // Global aggregation.
+    testAggregations(
+        vectors,
+        {},
+        {min(c1), max(c1)},
+        fmt::format("SELECT min({}), max({}) FROM tmp", c1, c1));
+
+    // Group by aggregation.
+    testAggregations(
+        [&](auto& builder) {
+          builder.values(vectors).project({"c0 % 10", "c1"});
+        },
+        {"p0"},
+        {min(c1), max(c1)},
+        fmt::format(
+            "SELECT c0 % 10, min({}), max({}) FROM tmp GROUP BY 1", c1, c1));
+
+    // Masked aggregations.
+    auto minMaskedAgg = min(c1) + " filter (where mask)";
+    testAggregations(
+        vectors,
+        {},
+        {minMaskedAgg},
+        fmt::format("SELECT min({}) filter (where mask) FROM tmp", c1));
+
+    testAggregations(
+        [&](auto& builder) {
+          builder.values(vectors).project({"c0 % 10", "c1", "mask"});
+        },
+        {"p0"},
+        {minMaskedAgg},
+        fmt::format(
+            "SELECT c0 % 10, min({}) filter (where mask) FROM tmp GROUP BY 1",
+            c1));
+
+    auto maxMaskedAgg = max(c1) + " filter (where mask)";
+    testAggregations(
+        vectors,
+        {},
+        {maxMaskedAgg},
+        fmt::format("SELECT max({}) filter (where mask) FROM tmp", c1));
+
+    testAggregations(
+        [&](auto& builder) {
+          builder.values(vectors).project({"c0 % 10", "c1", "mask"});
+        },
+        {"p0"},
+        {maxMaskedAgg},
+        fmt::format(
+            "SELECT c0 % 10, max({}) filter (where mask) FROM tmp GROUP BY 1",
+            c1));
+
+    // Encodings: use filter to wrap aggregation inputs in a dictionary.
+    testAggregations(
+        [&](auto& builder) {
+          builder.values(vectors)
+              .filter("c0 % 2 = 0")
+              .project({"c0 % 11", "c1"});
+        },
+        {"p0"},
+        {min(c1), max(c1)},
+        fmt::format(
+            "SELECT c0 % 11, min({}), max({}) FROM tmp WHERE c0 % 2 = 0 GROUP BY 1",
+            c1,
+            c1));
+
+    testAggregations(
+        [&](auto& builder) { builder.values(vectors).filter("c0 % 2 = 0"); },
+        {},
+        {min(c1), max(c1)},
+        fmt::format(
+            "SELECT min({}), max({}) FROM tmp WHERE c0 % 2 = 0", c1, c1));
+  }
+};
+
+TEST_F(MinMaxAggregationTest, tinyint) {
```

Comment:
I wonder if we need to copy tests for all types if only timestamp is touched.

Reply:
Presto also regiesters max_n / min_n functions, so we can't reuse it and need to have onw register function for Spark. So I think we need to test all types, BTW I refactor the test to combine min and max for one type to one case.